index.js

```diff
@@ -379,24 +380,39 @@ function Argv (processArgs, cwd) {
 
   var versionOpt = null
   self.version = function (opt, msg, ver) {
-    if (arguments.length === 1) {
+    if (arguments.length === 0) {
+      ver = guessVersion()
+      opt = 'version'
+    } else if (arguments.length === 1) {
       ver = opt
-      versionOpt = 'version'
-      msg = usage.deferY18nLookup('Show version number')
+      opt = 'version'
     } else if (arguments.length === 2) {
-      versionOpt = opt
       ver = msg
-      msg = usage.deferY18nLookup('Show version number')
-    } else {
-      versionOpt = opt
     }
 
+    versionOpt = opt
+    msg = msg || usage.deferY18nLookup('Show version number')
+
     usage.version(ver || undefined)
     self.boolean(versionOpt)
     self.describe(versionOpt, msg)
     return self
   }
 
+  function guessVersion () {
+    var obj = readPkgUp.sync({
+      cwd: path.resolve(__dirname, '../')
+    })
+
+    // fallback to process.cwd() this
+    // solves the case of symlinked modules.
+    if (!obj.pkg) {
+      obj = readPkgUp.sync()
+    }
```

Comment:
What's the use-case for the symlink? Is it for when yargs is a symlinked global dependency (e.g. using `npm link`)? I'm just concerned that falling back to cwd here might return different, unexpected results if a consumer has symlinked their yargs dependency and relies on this functionality unintentionally - in which case they'd be better off handing yargs an explicit version value.

Typically, the only symlink involved in a real-world global install of a consumer would the bin executable on the user's PATH, which would point to a normal `node_modules` install such that the first `readPkgUp` call would work.